app/assets/javascripts/discourse/app/components/d-button.gjs

```diff
@@ -127,7 +128,7 @@ export default class DButton extends GlimmerComponentWithDeprecatedParentView {
           if (forwardEvent) {
             actionVal(actionParam, event);
           } else {
-            actionVal(actionParam);
+            runAfterFramePaint(() => actionVal(actionParam));
           }
         }
```

Comment:
I think we should probably wrap this whole thing in `runAfterFramePaint`. As it is, we're only handling one possible way of passing an action to DButton

Reply:
Fair. @tyb-talks was also mentioning about the other cases.

I originally chose to isolate in this particular case because the usages were obvious, but ok to apply it to all cases.

(Also, I was considering adding it to `sendWidgetAction` as well but I will look for usages first.)